ark/spLDA/processing.py

```diff
@@ -63,24 +52,153 @@ def format_cell_table(cell_table, markers=None, clusters=None, fovs="all_fovs"):
+    else:
+        neighborhood_feature_fn = functools.partial(func_type[feature_by])
+
+    # Featurize FOVs
+    feature_sample = {k: v for (k, v) in cell_table.items() if
+                      k in cell_table["fovs"].tolist()}
+    featurized_fovs = featurize_samples(feature_sample,
+                                        neighborhood_feature_fn,
+                                        radius=radius,
+                                        is_anchor_col=cell_index,
+                                        x_col='x',
+                                        y_col='y',
+                                        n_processes=n_processes,
+                                        include_anchors=True)
+
+    if train_split:
+        all_sample_idxs = featurized_fovs.index.map(lambda x: x[0])
+        _sets = train_test_split(featurized_fovs, test_size=1. - train_frac,
+                                 stratify=all_sample_idxs)
+        train_features_fraction, _ = _sets
+    else:
+        train_features_fraction = None
+
+    return {"featurized_fovs": featurized_fovs,
+            "train_features": train_features_fraction}
+
+
+def create_difference_matrices(cell_table, features, training=True,
+                               inference=True):
+    """Constructs the difference matrices used for training and inference for
+    each field of view in the formatted cell table.
 
+    Args:
+        cell_table (dict, pd.DataFrame):
+            A formatted cell table for use in spatial-LDA analysis.
+            Specifically, this is the output from
+            format_cell_table().
+        features (dict, pd.DataFrame):
+            A featurized cell table and training split. Specifically, this is
+            the output from featurize_cell_table().
+        training (bool):
+            If True, create the difference matrix for running training
+            algorithm.  One or both of training and inference
+            must be True.
+        inference (bool):
+             If True, create the difference matrix for running inference
+             algorithm.
 
+    Returns:
+        dict:
 
+        - A dictionary containing the training and inference difference
+        matrices.
+    """
+    cell_table = {k: v for (k, v) in cell_table.items() if
+                  k not in ["fovs", "markers", "clusters"]}
+    # check args function here
+    if training:
+        train_diff_mat = make_merged_difference_matrices(
+            sample_features=features["train_features"],
+            sample_dfs=cell_table,
+            x_col="x", y_col="y")
+    else:
+        train_diff_mat = None
+
+    if inference:
+        inference_diff_mat = make_merged_difference_matrices(
+            sample_features=features["featurized_fovs"],
+            sample_dfs=cell_table,
+            x_col="x", y_col="y")
+    else:
+        inference_diff_mat = None
+
+    return {"train_diff_mat": train_diff_mat,
+            "inference_diff_mat": inference_diff_mat}
```

Comment:
Could be condensed as:
```
arg_settings = {
    "train_diff_mat" : (training, "train_features"),
    "inference_diff_mat": (inference, "featureized_fovs"),
}

return {
    k: None if not v[0] else make_merged_difference_matrices(
        sample_features=features[v[1]],
        sample_dfs=cell_table,
        x_col="x", y_col="y")
    for (k, v) in arg_settings
}
```
but with less nasty tabbing

Reply:
I think this is a bit too fancy for my tastes. It's about half as long, but takes twice as long to figure out what's going on.

Reply:
This works but it's probably best to assign this to a separate variable first before returning. I like to keep my return statements simple 😂.